client/dashboard/utils/link.ts

```diff
@@ -12,3 +12,18 @@ import config from '@automattic/calypso-config';
 export function wpcomLink( path: string ) {
 	return `${ config( 'wpcom_url' ) }${ path }`;
 }
+
+/**
+ * This function returns the link to the reauth page.
+ *
+ * Currently, the dashboard run in either Calypso or Dashboard environment. When it comes to the redirect URL:
+ * - For Calypso, the reauth page is hosted on the same origin so we use window.location.pathname.
+ * - For Dashboard, the reauth page isn't hosted on the same origin so we use window.location.href.
+ */
+export function reauthRequiredLink() {
+	const wpcomUrl = config( 'wpcom_url' );
+	const isSameOrigin = window.location.origin === wpcomUrl;
```

Comment:
Should we normalize the origins?
```js
const wpcomUrl = config( 'wpcom_url' );
const wpcomOrigin = new URL( wpcomUrl ).origin;
const isSameOrigin = window.location.origin === wpcomOrigin;
```

Reply:
Maybe just `wpcomUrl.startsWith( window.location.origin )`?